### ISSUE TITLE
OS X overlay does not work in Minecraft

### ISSUE THREAD

**Issue by mkrautz** [mumble-voip/mumble#1947]
nzw1 reports on IRC that the OS X overlay does not work in Minecraft.

**Comment by nzw1** [comment]
I can't see any overlay install new version and all still doesn't work any ideas thank you

**Comment by mkrautz** [comment]
@nzw1 If you had the option to upgrade the overlay in Preferences -> Overlay after upgrading to the snapshot build, and you did that, then you might need to log out of your OS X user and log back in.

**Comment by nzw1** [comment]
I have tried that still doesn't work

**Comment by nzw1** [comment]
tried deleting

**Comment by ancarda** [comment]
I have been able to reproduce this (Mumble 1.3.3 on macOS 10.15.7 using Minecraft Java Edition 1.16.4).

I was not able to find a configuration that worked. I tried setting the overlay feature to different modes and also checked System Preferences; Mumble had access to the Minecraft Launcher.

Interestingly, I was never prompted to give Mumble access to the `java` process that the launcher spawns. That particular process is located on disk here:

    ~/Library/Application Support/minecraft/runtime/jre-x64/jre.bundle/Contents/Home/bin/java

Furthermore, I was not able to select this process in the file browser to add to the overlay whitelist. Perhaps this problem is caused, in part, by how Minecraft is packaged and run on macOS? Normally, applications would be inside `.app` directories, where the binary is at `Application.app/Contents/MacOS/application`.

Does anyone know if the overlay works with other games on macOS? And does the overlay work with Minecraft Java Edition on other operating systems?

**Comment by Krzmbrzl** [comment]
@TerryGeng did you by any chance use/test the overlay on macOS at some point?

**Comment by TerryGeng** [comment]
@Krzmbrzl I didn't really try overlay, but I just gave it a brief shot. Mumble 1.3.3 with MacOS 10.15.7 (newest Catalina). What I have done:
- Installed Overlay,
- Enabled Overlay,
- Set the Exception Mode to be blacklist and had no program blacklisted, so I think it basically meant everything is allowed,
- Start games.

I first tried Minecraft, as @ancarda said, nothing prompted up, and overlay not shown as well.

I then tried some steam games, i.e. Overcook 2. This time MacOS did ask me for my permission, but again, nothing shows up after the game started.

Interestingly, I noticed a warning message 
```
2020-11-18 08:38:21.459 Mumble[16994:5262373] warning: failed to get scripting definition from /Users/terrygeng/Library/Application Support/Steam/steamapps/common/Overcooked! 2/Overcooked2.app; it may not be scriptable.
```
when I was starting Overcook. A search in the codebase gave me nothing. This line doesn't look like the output of the log engine of the Qt. I'm not sure if it is related to this problem.

**Comment by ancarda** [comment]
That warning gives me a bad feeling System Integrity Protection may be getting in the way. I can run some more tests with a few random Steam games this weekend. Sadly I don't have the disk space for it right now

**Comment by ancarda** [comment]
I tried [American Truck Simulator](https://store.steampowered.com/app/270880/American_Truck_Simulator/) and [Universe Sandbox](https://store.steampowered.com/app/230290/Universe_Sandbox/) just now. Both with and without SIP, I never saw the overlay.

However, disabling System Integrity Protection was enough to get the overlay to show on Minecraft! So, there's a possible workaround.

However, I will say the overlay looks very small and seems to be limited to the bottom left corner/quadrant. That could be some kind of scaling issue as I do have a High-DPI ("Retina") Mac, so perhaps it doesn't scale up properly?

**Comment by Krzmbrzl** [comment]
I have never really used the overlay nor did I work much on its code so I can onky give limited insights here.
Scaling issues do seem very plausible though :point_up:

**Comment by ZeroAbility** [comment]
Was this resolved in 1.4?
